actors.md

```diff
@@ -580,6 +580,38 @@ func SubmitPost(proofs []PoStProof, faults []FaultSet, recovered BitField, done
 	miner.NextDoneSet = done
 	miner.ArbitratedDeals.Clear()
 }
+
+func ValidateFaultSets(faults []FaultSet, recovered, done BitField) bool {
+	var aggregate BitField
+	for _, fs := range faults {
+		aggregate.Combine(fs.BitField)
+	}
+
+	// all sectors marked recovered must have actually failed
+	if !recovered.IsSubsetOf(aggregate) {
+		return false
+	}
+
+	// the done set cannot intersect with the aggregated faults
+	// you can't mark a fault as 'done'
+	if aggregate.Intersects(done) {
+		return false
+	}
+
+	for _, bit := range aggregate.Bits() {
+		if !miner.HasSectorByID(bit) {
```

Comment:
It's my belief that this `aggregate` includes indices into the miner's `SectorSet`. I think that it will be required that we build something which maps a miner's sector id to an index in the `SectorSet`. If so, I think this pseudocode should be updated to call that function and that the function's output should be passed to `miner.HasSectorByID`.

Reply:
Note: The above recommendation assumes that the sector IDs cannot be used as indices into the `SectorSet`. The current version of the specification permits miners to pick their own sector ids - so there could be a large gap in between any two sector IDs. Unsure what impact, if any, this would have on the `SectorSet`.

Reply:
sector IDs are indexes into the sector set, having gaps is allowable, and should add minimal overhead to the bitfields. This is discussed in #116 (giant issue i know, we should write up a summary)

Reply:
FYI I'm working on a summary of the Bitfield and SectorSet design discussions as tracked [here](https://github.com/filecoin-project/go-filecoin/issues/2824)